tripy/tripy/frontend/module/sequential.py

```diff
@@ -0,0 +1,186 @@
+# limitations under the License.
+from typing import Union, Tuple, Dict, Iterator, Any
+from dataclasses import dataclass
+import copy
+
+from tripy import export
+from tripy.frontend.module import Module
+
+
+@export.public_api(
+    document_under="modules/sequential.rst",
+    autodoc_options=[
+        ":exclude-members: __getattr__",
+    ],
+)
+@dataclass
+class Sequential(Module):
+    r"""
+    A module to stack multiple layers or modules in a sequential order. The `Sequential`
+    container can accept either a list of modules or a dictionary of named modules. Modules are
+    added in the order they are passed, and each is called sequentially during the forward pass.
+    """
+
+    def __init__(self, *modules: Union[Module, Dict[str, Module]]) -> None:
+        r"""
+        Args:
+            *modules: The modules to include in the sequence.
+                Can be passed as individual positional arguments or as a single dictionary of named modules.
+
+        .. code-block:: python
+            :linenos:
+            :caption: Sequential with Positional Arguments
+
+            model = tp.Sequential(tp.Linear(1, 3), tp.Linear(3, 2))
+
+            input = tp.Tensor([1.0])
+            output = model(input)
+
+        .. code-block:: python
+            :linenos:
+            :caption: Sequential with a Dictionary
+
+            model = tp.Sequential({'layer1': tp.Linear(1, 3), 'layer2': tp.Linear(3, 2)})
+
+            input = tp.Tensor([1.0])
+            output = model(input)
+        """
+        super().__init__()
+        self.modules = {}
+
+        # Disallow mixing modules/dicts
+        if any(isinstance(m, dict) for m in modules) and any(not isinstance(m, dict) for m in modules):
+            raise ValueError("Cannot mix dictionaries and individual modules in Sequential.")
+
+        if len(modules) == 1 and isinstance(modules[0], dict):
+            self.modules = copy.copy(modules[0])
+        else:
+            for idx, module in enumerate(modules):
+                self.modules[str(idx)] = module
+
+    def __call__(self, input: "tripy.Tensor") -> "tripy.Tensor":
+        r"""
+        Defines the forward pass by applying each module in the container sequentially to `input`
+
+        Args:
+            input: The input tensor to pass through the sequence of modules.
+
+        Returns:
+            The output tensor after passing through each module in sequence.
+        """
+        for module in self.modules.values():
+            input = module(input)
+        return input
+
+    def __getattr__(self, name) -> Any:
+        """
+        Custom __getattr__ to search both in `modules` dictionary and in other attributes. This is for handling
+        `module = operator.attrgetter(child_name)(module)` calls in tripy/frontend/module/module.py:load_state_dict
+        """
+        # Check if `name` is a key in the modules dictionary
+        if "modules" in self.__dict__ and name in self.modules:
```

Comment:
Is the first part of the check necessary? Wouldn't the modules member always be defined?

Reply:
remvoed